PokeAlarm/Events/QuestEvent.py

```diff
@@ -0,0 +1,76 @@
+# Standard Library Imports
+import datetime
+# 3rd Party Imports
+# Local Imports
+from PokeAlarm import Unknown
+from . import BaseEvent
+from PokeAlarm.Utils import get_gmaps_link, get_applemaps_link, \
+    get_waze_link, get_dist_as_str
+
+
+class QuestEvent(BaseEvent):
+    """ Event representing the discovery of a PokeStop. """
+
+    def __init__(self, data):
+        """ Creates a new Quest Event based on the given dict. """
+        super(QuestEvent, self).__init__('quests')
+        check_for_none = BaseEvent.check_for_none
+
+        # Identification
+        self.stop_id = data['pokestop_id']
+        self.stop_name = check_for_none(
+            str, data.get('pokestop_name') or data.get('name'),
+            Unknown.REGULAR)
+        self.stop_image = check_for_none(
+            str, data.get('pokestop_url') or data.get('url'), Unknown.REGULAR)
+
+        # Location
+        self.lat = float(data['latitude'])
+        self.lng = float(data['longitude'])
+
+        # Completed by Manager
+        self.distance = Unknown.SMALL
+        self.direction = Unknown.TINY
+
+        # Used to reject
+        self.name = self.stop_id
+        self.geofence = Unknown.REGULAR
+        self.custom_dts = {}
+
+        # Quest Details
+        self.quest = data['quest']
+        self.reward = data['reward']
+        self.expiry = datetime.datetime.now().strftime("%d/%m/%Y 23:59")
```

Comment:
It's not sent in the webhook
Quests always expire at midnight so there's not much point to send it.

Reply:
Without sending the date, there is the possibility of introducing a bug where a webhook gets sent before midnight, is delayed for some reason, and isn't received until after. When PA displays the info it would then be incorrect. It's probably not a huge priority, but something to consider.

Reply:
Research Tasks change either at midnight or when Niantic starts/ends events that also changes the research task pool. 

For community days, this means research tasks will change 3 times during the day: midnight, 11 local time and finally back to the originals (src needed) at 14 local time.

For days that end or start events (like Halloween event) this means tasks rotate some arbitary time of the day, which you cannot easily know, since event times are special.

I don't see the point of implementing an expire_time at all for any purpose. The only reason why you'd want it is if PA queue is large and you somehow want PA to ignore too old data. This could be better done with a `datetime.datetime.now()` and check if it's still the same day.

Reply:
I still think this field should be sourced from the scanner side. I disagree that it doesn't have purpose - the field seems to have value for deduping, resolving the issues with delays, and with areas around horizons of time change. Presumably it's somewhat relevant for the user to know how long they have to pick up the quest as well?

Reply:
I can see how this is an issue on both sides, but using `datetime.datetime.now()` would force an expire time when it's possible the owner of the manual submissions may be aware of the varying expire times for quests on days such as community days. Likewise, there probably won't be much of a queue at to really have a need for an expire time field.

From where I'm sitting there's no easy way to solve both issues without adding another webhook field -- `override` or something like that -- which will tell PA to just ignore the cache.

Reply:
The protos in the real game suggests that they use an "updated" timestamp to figure out if a research task is the same or not. 
If the field is a "was discovered at TT" or "expected to expire at TT" doesn't make much difference to me, I'm just waffling out my own thoughts here.  Either way, if this field stays it will be more work for any tools wanting to send data to PA